packages/models-library/src/models_library/projects.py

```diff
@@ -40,33 +52,73 @@ class Project(BaseModel):
         description="longer one-line description about the project",
         examples=["Dabbling in temporal transitions ..."],
     )
-    # NOTE: str is necessary because HttpUrl will not accept and empty string and the
-    # frontend sometimes sends this empty string, which is removed by the validator
-    thumbnail: Union[str, HttpUrl] = Field(
+    thumbnail: Optional[HttpUrl] = Field(
         ...,
         description="url of the project thumbnail",
         examples=["https://placeimg.com/171/96/tech/grayscale/?0.jpg"],
     )
 
-    # Ownership and Access  (SEE projects_access.py)
-    prjOwner: EmailStr = Field(..., description="user email")
-    accessRights: Dict[GroupID, AccessRights] = Field(
-        ...,
-        description="object containing the GroupID as key and read/write/execution permissions as value",
+    creation_date: datetime = Field(...)
+    last_change_date: datetime = Field(...)
+
+    # Pipeline of nodes (SEE projects_nodes.py)
+    workbench: Workbench = ...
+
+    @validator("thumbnail", always=True, pre=True)
+    @classmethod
+    def convert_empty_str_to_none(v):
+        if isinstance(v, str) and v == "":
+            return None
+        return v
+
+
+class ProjectAtDB(ProjectCommons):
+    # specific DB fields
+    id: int = Field(..., description="The table primary index")
+    project_type: ProjectType = Field(..., alias="type", description="The project type")
+    prj_owner: Optional[int] = Field(..., description="The project owner id")
+    published: Optional[bool] = Field(
+        False, description="Defines if a study is available publicly"
     )
 
+    @validator("project_type", pre=True)
+    @classmethod
+    def convert_sql_alchemy_enum(cls, v):
+        if isinstance(v, Enum):
+            return v.value
+        return v
+
+    class Config:
+        orm_mode = True
+        use_enum_values = True
+
+
+class Project(ProjectCommons):
+    # NOTE: This is the pydantic pendant of project-v0.0.1.json used in the API of the webserver/webclient
+    # NOT for usage with DB!!
+
+    # Ownership and Access  (SEE projects_access.py)
+    prj_owner: EmailStr = Field(..., description="user email", alias="prjOwner")
+
     # Timestamps   TODO: should we use datetime??
-    creationDate: str = Field(
+    creation_date: str = Field(
         ...,
         regex=DATE_RE,
         description="project creation date",
         examples=["2018-07-01T11:13:43Z"],
+        alias="creationDate",
```

Comment:
In all web API models, we should use by default ``alias_generator=snake_to_camel`` in the ``Config``?  (Search ``alias_generator`` in ``web/server`` for samples)

Reply:
ok, I thought the aliases had to be set explicitely now. will try